spec/report_spec.rb

```diff
@@ -1155,6 +1155,62 @@ def gloops
     }
   end
 
+  it "filters params from all payload hashes if they are added to redacted_keys as a string" do
+    Bugsnag.configuration.redacted_keys << "other_data"
+
+    Bugsnag.notify(BugsnagTestException.new("It crashed")) do |report|
+      report.add_metadata(:request, {
+        params: {
+          password: "1234",
+          other_password: "123456",
+          other_data: "123456",
+          more_other_data: "123456",
+          abc: "xyz"
+        }
+      })
+    end
+
+    expect(Bugsnag).to have_sent_notification{ |payload, headers|
+      event = get_event_from_payload(payload)
+      expect(event["metaData"]).not_to be_nil
+      expect(event["metaData"]["request"]).not_to be_nil
+      expect(event["metaData"]["request"]["params"]).not_to be_nil
+      expect(event["metaData"]["request"]["params"]["password"]).to eq("[FILTERED]")
+      expect(event["metaData"]["request"]["params"]["other_password"]).to eq("[FILTERED]")
```

Comment:
Shouldn't the match for string keys be exact, so not match "other_password" or "more_other_data"?

(Also, should we assert that in the cleaner spec?)

Reply:
It is exact — `other_password` & `password` are being filtered by the default `meta_data_filters`

`other_data` is filtered by `redacted_keys` and `more_other_data` is not, which proves it's exact

Reply:
Ah - got it now. OK, I guess that simplifies significantly in major release.